TextToSpeech/impl/TTSManager.cpp

```diff
@@ -50,19 +49,20 @@ TTSManager::~TTSManager() {
 }
 
 TTS_Error TTSManager::enableTTS(bool enable) {
-
-    if(m_ttsEnabled != enable) {
-        m_ttsEnabled = enable;
-        TTSLOG_INFO("TTS is %s", enable ? "Enabled" : "Disabled");
-        m_callback->onTTSStateChanged(m_ttsEnabled);
-        m_speaker->ensurePipeline(m_ttsEnabled);
-    }
-
+    bool updated = false;
+    updated |= m_defaultConfiguration.setEnabled(enable);
+    TTSLOG_INFO("TTS is %s", enable ? "Enabled" : "Disabled");
+    if(updated)
+        m_defaultConfiguration.updateConfigStore();
+    if(!m_defaultConfiguration.enabled())
+        shut(0);
+    m_callback->onTTSStateChanged(m_defaultConfiguration.enabled());
```

Comment:
onTTSStateChanged() & ensurePipeline() should happen only if there is a state change